app/Helpers/GroupBalancedDivision.php

```diff
@@ -112,35 +112,38 @@ protected function assignNonDrinkers()
     {
         $nonDrinkerRegs = $this->registrations->where('drinks_alcohol', '=', false);
 
-        $nonDrinkerFillRates = $this->calcFillRate($nonDrinkerRegs);
+        // check if there are any non drinkers
+        if ($nonDrinkerRegs->count() > 0) {
+            $nonDrinkerFillRates = $this->calcFillRate($nonDrinkerRegs);
 
-        // If you can satisfy minNonDrinkers requirement with a balanced fill, do it first
-        if (array_sum($nonDrinkerFillRates[min(array_keys($nonDrinkerFillRates))]) > $this->minNonDrinkers) {
-            $this->assignBalanced($nonDrinkerRegs, $nonDrinkerRegs);
-        }
-
-        // Assign yet unassigned non-drinkers
-        $nonDrinkerRegs = $nonDrinkerRegs->where('group_id', '=', null)
-            ->shuffle();
+            // If you can satisfy minNonDrinkers requirement with a balanced fill, do it first
+            if (array_sum($nonDrinkerFillRates[min(array_keys($nonDrinkerFillRates))]) > $this->minNonDrinkers) {
+                $this->assignBalanced($nonDrinkerRegs, $nonDrinkerRegs);
+            }
 
-        // If chunking by minNonDrinkers would give more chunks than groups, increase chunk size by 1 until it fits
-        $nonDrinkersPerGroup = $this->minNonDrinkers;
-        while (($nonDrinkerRegs->count() / $nonDrinkersPerGroup) > $this->groups->count()) {
-            $nonDrinkersPerGroup++;
-        }
-        $chunks = $nonDrinkerRegs->chunk($nonDrinkersPerGroup);
+            // Assign yet unassigned non-drinkers
+            $nonDrinkerRegs = $nonDrinkerRegs->where('group_id', '=', null)
+                ->shuffle();
 
-        // Assign each chunk to a group
-        foreach ($this->groups as $group) {
-            if ($chunks->count() == 0) {
-                return;
+            // If chunking by minNonDrinkers would give more chunks than groups, increase chunk size by 1 until it fits
+            $nonDrinkersPerGroup = $this->minNonDrinkers;
+            while (($nonDrinkerRegs->count() / $nonDrinkersPerGroup) > $this->groups->count()) {
+                $nonDrinkersPerGroup++;
             }
-            $chunk = $chunks->pop();
+            $chunks = $nonDrinkerRegs->chunk($nonDrinkersPerGroup);
 
-            foreach ($chunk as $registration) {
-                $registration->group_id = $group->id;
-                $registration->queue_position = null;
-                $registration->save();
+            // Assign each chunk to a group
+            foreach ($this->groups as $group) {
+                if ($chunks->count() == 0) {
+                    return;
+                }
+                $chunk = $chunks->pop();
+
+                foreach ($chunk as $registration) {
+                    $registration->group_id = $group->id;
+                    $registration->queue_position = null;
+                    $registration->save();
+                }
```

Comment:
Could you explain why you changes these lines for implementing a test for this function? :D (this is weird if you think about it)
Is this a bugfix in the algorithm?

Reply:
Sorry, now i saw your comment.

> GroupBalancedDivision: if-clause to check for non drinkers (ran into exception with no non drinkers)

Reply:
Does this maybe also fix #503 ?

Reply:
> Does this maybe also fix #503 ?

Maybe yeah. Nice catch

Reply:
I don't think so. I think the imbalance in this issue is because assigning non-drinkers takes priority over dividing users by courses equally. This could lead to said imbalance in some cases